src/kind.ts

```diff
@@ -0,0 +1,74 @@
+import { Job } from "@brigadecore/brigadier";
+
+export const kindJobImage = "radumatei/golang-kind:1.11-0.4";
+
+export class KindJob extends Job {
+    constructor(name: string, image?: string) {
+        if (image == undefined) {
+            image = kindJobImage;
+        }
+        super(name, image);
+
+        // kind needs to run as a privileged pod
+        this.privileged = true;
+
+        // kind needs to add the following volumeMounts to function properly
+        // the Brigade project must enable `allowMountHosts`
+        this.volumes = [
+            {
+                name: "modules",
+                hostPath: {
+                    path: "/lib/modules",
+                    type: "Directory"
+                }
+            },
+            {
+                name: "cgroup",
+                hostPath: {
+                    path: "/sys/fs/cgroup",
+                    type: "Directory"
+                }
+            },
+            {
+                name: "docker-graph-storage",
+                emptyDir: {}
+            }
+        ];
+        this.volumeMounts = [
+            {
+                name: "modules",
+                mountPath: "/lib/modules",
+                readOnly: true
+            },
+            {
+                name: "cgroup",
+                mountPath: "/sys/fs/cgroup"
+            },
+            {
+                name: "docker-graph-storage",
+                mountPath: "/var/lib/docker"
+            }
+        ];
+
+        // to add your own tasks to this job, use job.tasks.push()
+        this.tasks = [
+            "dockerd-entrypoint.sh &",
+            "sleep 20",
+            "kind create cluster",
+            `export KUBECONFIG="$(kind get kubeconfig-path)"`,
+            "kubectl cluster-info",
+
+            // this pod is runing inside a Kubernetes cluster
+            // unset environment variables pointing to the host cluster
+            // even if the service account is limited, and KUBECONFIG is properly set,
+            // some operations might still point to the in-cluster configuration
+            "unset $(env | grep KUBERNETES_ | xargs)",
+
+            // the time needed for the control plane to properly start is around 2 minutes
+            //
+            // TODO: use a kubectl plugin to wait for all pods of the control plane 
+            "sleep 150",
```

Comment:
The wait periods are kind of arbitrary here. We should use a more sane approach (such as a `kubectl` plugin for waiting resources.

Reply:
Specifically, we have to wait for the following:

```
kubectl get all -n kube-system
NAME                                             READY   STATUS    RESTARTS   AGE
pod/coredns-5c98db65d4-6kklh                     1/1     Running   0          18m
pod/coredns-5c98db65d4-jkr9h                     1/1     Running   0          18m
pod/etcd-kind-control-plane                      1/1     Running   0          17m
pod/kindnet-l7b42                                1/1     Running   1          18m
pod/kube-apiserver-kind-control-plane            1/1     Running   0          17m
pod/kube-controller-manager-kind-control-plane   1/1     Running   2          17m
pod/kube-proxy-k8fkc                             1/1     Running   0          18m
pod/kube-scheduler-kind-control-plane            1/1     Running   1          17m


NAME               TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                  AGE
service/kube-dns   ClusterIP   10.96.0.10   <none>        53/UDP,53/TCP,9153/TCP   18m

NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                 AGE
daemonset.apps/kindnet      1         1         1       1            1           <none>                        18m
daemonset.apps/kube-proxy   1         1         1       1            1           beta.kubernetes.io/os=linux   18m

NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/coredns   2/2     2            2           18m

NAME                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/coredns-5c98db65d4   2         2         2       18m
```

cc @dgkanatsios

Reply:
You know what, I think adding `--wait xxx` to the `kind create cluster` command is all we need: https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster

We could default to... `300s`?  Perhaps allow override from user?

Reply:
Could also expose simple binary letting the user choose whether or not to wait, e.g., `job.waitForKind = true|false`

Reply:
Thank you for that flag, didn't know about it!

I think this library should provide either:

- a default single-node cluster, and wait for the control plane to be ready
- allow passing your own cluster create command, with a config and everything.

Combining the two would only lead to confusion in my opinion, but I'm open to discussion here.

Reply:
Ah, that sounds reasonable. 👍